src/errors.rs

```diff
@@ -0,0 +1,33 @@
+// Copyright 2016 Mozilla
+//
+// Licensed under the Apache License, Version 2.0 (the "License"); you may not use
+// this file except in compliance with the License. You may obtain a copy of the
+// License at http://www.apache.org/licenses/LICENSE-2.0
+// Unless required by applicable law or agreed to in writing, software distributed
+// under the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR
+// CONDITIONS OF ANY KIND, either express or implied. See the License for the
+// specific language governing permissions and limitations under the License.
+
+use rusqlite;
+
+use mentat_query_parser;
+use mentat_sql;
+
+error_chain! {
```

Comment:
We have little to no choice.

We either define a stack of enums as deep as our call stack, or we use (or reimplement) `error_chain` to flatten the stack into a broad `ErrorKind` enum at compile time and at runtime.

Notably if you have a module that uses `error_chain`, the modules _it_ calls must either use `error_chain` or return `Result<T, std::error::Error>` — if you provide non-error types as the error branch of the `Result`, `error_chain` lacks the coercions it needs, and you have to do a bunch of work at every call site to bail with the right value.

The simplest thing to do is make _everything_ use `error_chain`. I dislike its lack of specificity (there are situations in which some error kinds cannot occur, but it can't model that easily), and I dislike the proliferation of `Result` types, and I dislike that it makes error handling less visible. But the alternative is to have a pretty decent chunk of the codebase devoted to modeling different error enums and `From` conversions between them.